### DISCUSSION TITLE
Replace a Conbee II

### DISCUSSION THREAD

**Discussion by wifiuk** [Koenkk/zigbee2mqtt#14861] [Devices]
Ive moved over from Deconz with a Conbee II to Z2M, however, I don't think the conbee is playing very nice with Z2M, I'm getting all sorts of lag and timeouts, and its not the network, or the wifi interference .

Now I want to get the best dongle that is the most stable, what is the 2022/2023 recommendation, maybe on that also incorporates thread/matter?

**Comment by wifiuk** [comment]
bump

**Comment by herolddev** [comment]
A lot of them.
The complete list is at the zigbee2mqtt.io official page, have a look at them here https://www.zigbee2mqtt.io/guide/adapters/
One adapter from Home Assistant [SkyConnect](https://community.home-assistant.io/t/home-assistant-skyconnect-usb-stick-announced-will-be-compatible-with-both-zigbee-and-thread-including-matter-chip-over-thread/433594) is not at zigbee2mqtt list yet as it is not released yet, but you can have a look on it as well.

While choosing the adapter, please note the description of the zigbee2mqtt adapter's page:
- adapters based on Texas Instruments **CC2531** chip - are not powerful and not recommended by Z2M
- adapters based on Silicon Labs **EFR32** chip - are in **experimental** support mode of Zigbee2MQTT. At the same time, adapters based on the EFR32 can potentially support Matter in the future. 
- adapters based on Texas Instruments **CC2652P** - are **recommended by Z2M** community and the only fully supported by Z2M (apart from CC2531 which is not powerful). 

You can choose either USB or LAN or those that support both - they are the so-called "hybrid type". Depending on your setup - you can choose the right one. The difference between USB, LAN, and Hybrid are:
- **USB** adapters have to be plugged into the USB port of your server where z2m is running. 
- **LAN** adapters - you can put an adapter remote wherever you want - even in another remote room/building (or you can run your HA/Z2M in a cloud and use the adapter at your home (which is impossible for USB adapters).
- "**Hybrid**" - you can use them in both USB or LAN mode, so a kind of "universal".  
LAN / Hybrid Zigbee adapters can either support POE powering or not.

So I would recommend to have a look at something based on CC2652P and that works in both USB and LAN modes, namely:

- [Gio-dot Z-Bee Duo with CC2652P](https://gio-dot.github.io/Z-Bee-Duo/).
- [ZigStar LAN Coordinator](https://zig-star.com/projects/zigbee-gw-lan/).
- [ZigStar PoE Coordinator](https://zig-star.com/projects/zigstar-olizig/).
- [SMLIGHT SLZB-06 Zigbee ethernet USB POE WiFi LAN adapter](https://smlight.tech/product/slzb-06/).

I am running personally the last one from the list, [SLZB-06](https://smlight.tech/product/slzb-06/).

**Reply by lux73** [reply]
[Slaesh's CC2652RB](https://slae.sh/projects/cc2652/) running smooth & reliable for me

**Comment by wifiuk** [comment]
How about something like this sonoff?

https://amzn.to/3Vhpvxd

**Reply by Matriciel** [reply]
This work very well when you flash firmware with the latest development version found here [https://github.com/Koenkk/Z-Stack-firmware/tree/develop/coordinator/Z-Stack_3.x.0/bin](url)

**Reply by wifiuk** [reply]
cool, hopefully, it will be delivered in the next few days, so will have a google on flashing it and will let you know how I get on.

I have also noticed since using z2mq and the conbee the devices on the battery are dying a lot quicker.

**Comment by michal-bartak** [comment]
> Ive moved over from Deconz with a Conbee II to Z2M, however, I don't think the conbee is playing very nice with Z2M, I'm getting all sorts of lag and timeouts, and its not the network, or the wifi interference .

I have ConbeeII with Z2M for about 2 years. And it works without a single issue.
Make sure you have up-to-date ConbeeII firmware installed.
Move the Conbee stick away from your computer using USB extension cable (known issue devices with USB3 onboard, especially with HDD connected)

**Reply by wifiuk** [reply]
yeah all of that has been done, its for sure the stick, it cant be anything else, will find out soon when the new stick arrives.

**Reply by SgtZapper** [reply]
Same as michalk-k I have ben running the ConbeeII with Z2M without any problem but as said positioning of the stick with a USB extension is crucial or I'll get timeouts and a bunch of errors all the time. I don't think I have updated the fw for a year or so now.

**Reply by Rick-van-Dam** [reply]
> > Ive moved over from Deconz with a Conbee II to Z2M, however, I don't think the conbee is playing very nice with Z2M, I'm getting all sorts of lag and timeouts, and its not the network, or the wifi interference .
> 
> I have ConbeeII with Z2M for about 2 years. And it works without a single issue.
> Make sure you have up-to-date ConbeeII firmware installed.
> Move the Conbee stick away from your computer using USB extension cable (known issue devices with USB3 onboard, especially with HDD connected)
> 
> 

If I use the new firmware it doesn't work at all. So iam using a older firmware and I seem to be stuck on it. It works okish but sometimes there is some random lag. That's with a usb extension cord.

Going to move to a sonoff as soon as iam mentally prepared to pair all my devices again.

**Reply by OrbitingOcelot** [reply]
> If I use the new firmware it doesn't work at all. So iam using a older firmware and I seem to be stuck on it. It works okish but sometimes there is some random lag. That's with a usb extension cord.

Confirmed! Same here, newer conbeeII firmware does not play nice with some of the devices I have, have to use 2 year old one.

**Comment by grimelke001** [comment]
Have used conbeeII, zigahzig 2652, sonoff dongle-E (efr32mg21) and now a few weeks a SLZB-06 and  all dongles before the last one had problems with my 20 devices iam running in Z2M. the SLZB-06 you can used it were you want! how you want! and its powerfull.....i live in a new builded house 2 years now, before with my conbeeII in an house from the 1960's i had never a problem. sinds i moved to the new house all the dongles ill tryed had a very long delay and were after a few days unavailebel. and now with the SLZB-06 it works under Z2M very fine, yes iam very content with this Dongle.
Thats my small expierince

**Comment by wifiuk** [comment]
the new stick has arrived, and not had a single error since, but only been a day so far

**Reply by wifiuk** [reply]
update, the stick most likely was the issue, this new stick has been rock solid

**Comment by Rick-van-Dam** [comment]
Switched to sonoff dongle p and in the last 3 days have had not a single error. Also but this might be a feeling I think everything is a bit more snappy, almost as if the light is turning off just before you press the button. Its nice.

That conbee 2 is now very suspicious. Maybe its time to put a warning there in the docs?

**Comment by wifiuk** [comment]
yeah I've had no issues since replacing the Conbee, I wont be going back to using it again.

**Comment by Hedda** [comment]
FYI, for reference also see this related discussion about migrating from one radio adapter type to any other radio adapter type -> https://github.com/Koenkk/zigbee2mqtt/discussions/16478

**Comment by protyposis** [comment]
I migrated from Conbee 2 to a Sonoff ZBDongle-P with the `zigpy` CLI to avoid repairing (https://github.com/Koenkk/zigbee2mqtt/discussions/16478#discussioncomment-4870120). My Conbee network suffered from randomly unresponsive lights (https://github.com/Koenkk/zigbee2mqtt/issues/9554) and latency issues. The unresponsiveness is fixed, the latency did not improve (but I might be a little sensitive there, e.g. when a light takes 200ms to turn on).

Anyway, the migration without repairing seems to have triggered the battery drain issue in Ikea switches (https://github.com/Koenkk/zigbee2mqtt/issues/3444). On the Conbee network I replaced the batteries ~ once a year, now they drain within a few days and require a repair afterwards to work again. So, the `zigby` migration path might not be super stable yet.

**Reply by Hedda** [reply]
FYI, if you flashed new firmware for IKEA TRÅDFRI remote recently then be aware that it is known to cause many issues on all Zigbee gateway platforms, so your problem does not have to do anything with zigpy-cli migration.

Regardless also be sure to upgrade the firmware on your Sonoff ZBDongle-P CC2652P-based dongle -> https://github.com/Koenkk/Z-Stack-firmware/tree/master/coordinator/Z-Stack_3.x.0/bin

**Reply by protyposis** [reply]
My problem is very likely related to the migration because the coordinator is the only thing that changed, and the behavior exists exactly since that change. I did not say that it is the fault of `zigpy`, only that I chose this migration route. After repairing the switches, the battery drain issue seems to be gone, so creating a fresh network and repairing all devices (as suggested by Z2M) would probably have avoided this issue. (I upgraded the coordinator firmware, and Ikea switches with both old 2.2.010 and the newest 2.3.079 firmware were affected.)

> new firmware for IKEA TRÅDFRI remote [...] is known to cause many issues

Could you provide a link to an issue description?

**Reply by nonamejuju** [reply]
Would love to have a detailed instruction of the steps to do this migration

**Reply by Hedda** [reply]
See related discussion here -> https://github.com/Koenkk/zigbee2mqtt/discussions/16478 and specifically the command here but replace `ezsp` with `deconz` -> https://github.com/Koenkk/zigbee2mqtt/discussions/16478#discussioncomment-4870120

If you need more detailed instructions than that then I do not recommend you attempt this and instead just re-pair all your devices.

Problem is zigbee-herdman library Zigbee2MQTT depends on is still missing deconz adapter backup -> https://github.com/Koenkk/zigbee-herdsman/issues/665

Until then can only do a partial migration from deconz using zigpy-cli as described in my comment above linked here -> https://github.com/Koenkk/zigbee2mqtt/discussions/16478#discussioncomment-4870120

Note! There is currently no migration path to migrate all settings and devices for deconz (ConBee/RaspBee). This procedure with zigpy-cli only helps migrate the Zigbee network (i.e. the paring with devices). This means you at least will not have to re-pair all your devices again with the znp adapter. However, none of the higher-level settings will be migrated, so elements such as device names will be lost and will need to be manually entered again. There is currently no migration path to migrate all the settings. At least most powered devices like lightbulbs should be automatically re-connect over time (you can speed this up by rebooting the device) but your battery-powered devices may need to be re-joined to the network for the migration to complete.

**Comment by Dam0s** [comment]
Does anyone know if the Silicon Labs chips are likely to be supported by z2m anytime soon? Been checking the site for several months now but no update from developer support. 
I bought a ZBDongle-E about a year ago to replace my Conbee II but it was really unstable. I was only able to pair about 20-30 devices until it became unusable. I was hoping that since the SkyConnect coordinator has been released with this chipset, that is might provide enough interest to get these officially supported. Would love to get off my Conbee II to see if it resolves connectivity with just a few child devices that drop off now and then. Also it's always stuggled to handle a lot of instructions at one time. e.g. turn on 12 bulbs at once with certain settings, fortunately z2m "groups" help with this. Generally the Conbee is doing a pretty good job with 126 devices running for a year and half now but hoping these new coordinators will do a better job.

**Reply by Hedda** [reply]
> Does anyone know if the Silicon Labs chips are likely to be supported by z2m anytime soon? Been checking the site for several months now but no update from developer support.

Silabs EmberZNet NCP adapters do work in Zigbee2MQTT (it uses the [ezsp adapter](https://github.com/Koenkk/zigbee-herdsman/tree/master/src/adapter/ezsp) in the zigbee-herdsman library that Zigbee2MQTT depends on), they are however [not yet listed as recommended (and still listed under experimental)](https://www.zigbee2mqtt.io/guide/adapters/) because the [ezsp adapter](https://github.com/Koenkk/zigbee-herdsman/tree/master/src/adapter/ezsp) in the zigbee-herdsman library is still missing backup and restore feature, see TODO -> https://github.com/Koenkk/zigbee-herdsman/issues/319

> I bought a ZBDongle-E about a year ago to replace my Conbee II but it was really unstable. 

I have seen that most people are reporting that the [ezsp adapter](https://github.com/Koenkk/zigbee-herdsman/tree/master/src/adapter/ezsp) has been relatively very stable for the last year, but regardless of which adapter is used it is still always highly recommended that every user proactively take those actions and try to follow all best practice tips to see if the problem symptoms go away before they even consider troubleshooting any issue deeper.

So do read and follow this regardless of the state of your existing set-up -> https://community.home-assistant.io/t/guide-for-zigbee-interference-avoidance-and-network-range-coverage-optimization/515752

Zigbee can be very finicky and demanding if not set up optimally for your specific environment, so for starters, I always suggest being sure to add a few "known good" Zigbee Router devices (as having many "known good" Zigbee Router devices are available at all time is the main key to a stable Zigbee network mesh), and taking some basic actions to avoid interference is important too.

As for recommended "known good" Zigbee Router devices I normally recommend beginners who want a quick and easy solution that is good enough for this to buy and a few of "IKEA Trådfri Signal Repeater" (also I read that the "Aeotec Range Extender Zi" works well to), or for those wanting the best and willing to do a DIY solution to buy a few "Sonoff ZBDongle-E" or "Sonoff ZBDongle-P" adapters and re-flash them with Zigbee Router firmware then power them with USB-changes to create great dedicated Zigbee Router devices, (colloquially I heard that the "Sonoff ZBDongle-E" outperfoms the "Sonoff ZBDongle-P" when used as a such dedicated Zigbee repeater/extender). 

Using newer Zigbee Coordinator adapters with later firmware can also resolve or hide many issues, though they may also introduce new bugs, so it can be a gamble when choosing to use more mature firmware or the latest cutting-edge firmware. Here experimenting is the only option as it will depend on what exact devices you have and how well the firmware on those devices works with different Zigbee Coordinator firmware.

Anyway, for better or worse, a great Zigbee network mesh set-up does initially require some excessive care, precision, and attentive effort for optimal operation in the long run. However, once you have those basics in place, then it should just work automagically, (unless you add some rouge device with buggy firmware to your Zigbee network mesh that does not pass along all messages from its neighbour devices).
